stacks-signer/src/stacks_client.rs

```diff
@@ -49,11 +93,41 @@ impl From<&Config> for StacksClient {
                 config.stackerdb_contract_id.clone(),
             ),
             stacks_private_key: config.stacks_private_key,
+            stacks_address: config.stacks_address,
             slot_versions: HashMap::new(),
+            http_origin: format!("http://{}", config.node_host),
+            tx_version: TransactionVersion::Testnet,
+            chain_id: Network::Testnet.to_chain_id(),
+            stacks_node_client: reqwest::blocking::Client::new(),
         }
     }
 }
 
+/// Trait used to make interact with Clarity contracts for use in the signing process
+pub trait StacksContractCallable {
+    /// Submits a transaction to a node RPC server
+    fn submit_tx(&self, tx: Vec<u8>) -> Result<String, ClientError>;
+
+    /// Call read only tx
+    fn read_only_contract_call(
+        &self,
+        contract_addr: &StacksAddress,
+        contract_name: &str,
```

Comment:
We have specific containers for contract names, function names, and function arguments, since they have to use limited character sets and since not all strings are valid.  You should use `ContractName` and `ClarityName` here instead of a bare `&str`.  Also, `function_args` should be `&[clarity::vm::Value]` in both cases.

Reply:
Quick note, can you create a valid "StacksAddress" from the pox contract address? My initial attempts have failed. The reason I changed this to String was to accommodate this. However, perhaps I am just doing it wrong?

Reply:
Sure -- `QualifiedContractIdentifier::parse("ST000000000000000000002AMW42H.pox-3").unwrap()` should work.

Reply:
Yes sorry I wasn't clear/ I am saying that does not work 
 'called `Result::unwrap()` on an `Err` value: InvalidContractAddress("\"ST000000000000000000002AMW42H")',

Reply:
Sorry, I was not clear. This is what I already tried and it results in 
```thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Runtime(ParseError("Invalid principal literal: base58ck checksum 0x51104e95 does not match expected 0x31287a31"), None)', stacks-signer/src/stacks_client.rs:315:83```

This is the same result as trying to parse ST000000000000000000002AMW42H as a StacksAddress.